### DISCUSSION TITLE
Throw a warnings if unwanted file is being bundled to the frontend.

### DISCUSSION THREAD

**Discussion by arthurfiorette** [vercel/next.js#34400] [Ideas]
### Describe the feature you'd like to request

I know that this isn't a best practice or even optimal. But there's many people out there _(including myself, sometimes)_ that adds functions or values that are not meant to be bundled to the frontend, either by containing secret informations or business logic.

Next could add an option to warn (or throw an error) if the file is being bundled to the frontend.

### Describe the solution you'd like

I thought that a comment would perfect suit this, like `prettier` and `eslint` does:

```ts
/* nextjs-ignore bundle */

export const myBackendFunc = () => {
  // ...
};

export const mySecret = ':)';
```

And, when importing this file in a file that is going to be bundled in the frontend code, nextjs will emit an error (or warning) like:

```
error - File /lib/example.ts cannot be bundled. But it was imported from /pages/index.ts
```

A while ago I found a playground that shows us what is extracted from the page files to be in the package. So I think importing some file with this comment but just using it inside server side functions like getServerSideProps shouldn't throw this warning.

Also, if someone has this playground link, please send it again <3

### Describe alternatives you've considered

Well, going in the same way as `prettier` and `eslint`, this could be extented to be a property in the `next.config.js` or even create a `.nextjsignore` file.

**Comment by balazsorban44** [comment]
Hi, this has been discussed in detail at https://github.com/vercel/next.js/discussions/32874

You should not use anything else than environment variables for secret data
